src/handlers.rs

```diff
@@ -364,35 +364,44 @@ pub async fn handle_direct_message(
             }
         };
 
-        // Check if the target peer exists
-        let peer_exists = peer_names.values().any(|name| name == &to_name);
-        if !peer_exists {
-            ui_logger.log(format!(
-                "Peer '{}' not found. Use 'ls p' to see available peers.",
-                to_name
-            ));
-            return;
-        }
-
-        let direct_msg = DirectMessage::new(
-            PEER_ID.to_string(),
-            from_name,
-            to_name.to_string(),
-            message.to_string(),
-        );
+        // Check if the target peer exists and find their PeerId
+        let target_peer_id = peer_names.iter()
+            .find(|(_, name)| name == &&to_name)
+            .map(|(peer_id, _)| *peer_id);
+        
+        let target_peer_id = match target_peer_id {
+            Some(peer_id) => peer_id,
+            None => {
+                ui_logger.log(format!(
+                    "Peer '{}' not found. Use 'ls p' to see available peers.",
+                    to_name
+                ));
+                return;
+            }
+        };
 
-        let json = serde_json::to_string(&direct_msg).expect("can jsonify direct message");
-        let json_bytes = Bytes::from(json.into_bytes());
+        // Create a direct message request
+        let direct_msg_request = DirectMessageRequest {
```

Comment:
Should the from name and from peerid be validated against the actual connection to prevent spoofing.